service/storage_service.cc

```diff
@@ -1933,8 +1935,7 @@ future<> storage_service::node_ops_cmd_heartbeat_updater(node_ops_cmd cmd, node_
             auto ep = std::current_exception();
             slogger.warn("{}[{}]: Failed to send heartbeat: {}", ops, uuid, ep);
         }
-        int nr_seconds = 10;
-        while (!(*heartbeat_updater_done) && nr_seconds--) {
+        while (!(*heartbeat_updater_done) && std::chrono::steady_clock::now() < next) {
```

Comment:
As a follow up, this should be a single sleep_abortable
and heartbeat_updater_done should simply be an abort_source.